code/basic.js

```diff
@@ -0,0 +1,485 @@
+      furState: 'furry',
+    },
+  ]
+}
+
+// Contains info about the created kitten
+const kitten = {
+  _name: '',
+  _gender: '',
+  _environmentPreference: '', 
+  _ownerName: '',
+  furState: '',
+  imageSrc: '',
+  nameConfirmation: false,
+  set name(name) {
+    this.name = name.charAt(0).toUpperCase() + name.substr(1).toLowerCase()
+  },
+  set gender(data) {
+    if (data.toLowerCase() === 'girl') {
+      this._gender = data
+      console.log(this._gender)
+      this.pronoun = 'she'
+      this.possessivePronoun = 'her'
+      this.personalPronoun = 'her'
+    } else if (data.toLowerCase() === 'boy') {
+      this._gender = data
+      this.pronoun = 'he'
+      this.possessivePronoun = 'his'
+      this.personalPronoun = 'him'
+    } else if (data.toLowerCase() === "doesn't matter") {
+      this._gender = 'neutral'
+      this.pronoun = 'it'
+      this.possessivePronoun = 'its'
+      this.personalPronoun = 'it'
+    }
+  },
+  get gender() {
+    return this._gender
+  },
+  set environmentPreference(data) {
+    if (data.toLowerCase() === 'indoors') {
+      this._environmentPreference = 'indoors' 
+      this.activityPreference = 'cuddling up in the sofa'
+    } else if (data.toLowerCase() === 'outdoors'){
+      this._environmentPreference = 'outdoors' 
+      this.activityPreference = 'going on cute little adventures'
+    }
+  },
+  get environmentPreference() {
+    return this._environmentPreference
+  },
+  set name(data) {
+    this._name = data
+  },
+  get name() {
+    return this._name
+  },
+  set ownerName(data) {
+    this._ownerName = data.charAt(0).toUpperCase() + data.substr(1).toLowerCase()
+  },  
+  get ownerName() {
+    return this._ownerName
+  }
+}
+
+const playTypingAnimation = (time) => {
+  chat.innerHTML +=
+    `
+    <div id="lottie-animation">
+      <lottie-player src="https://assets5.lottiefiles.com/datafiles/t4b48ItXS0KF2VK/data.json"  background="transparent"  speed="0.6"  style="width: 50px; height: 40px;"  loop  autoplay></lottie-player>
+    </div>
+    `
+  chat.scrollTop = chat.scrollHeight
+  setTimeout(() => {
+    const lottie = document.getElementById('lottie-animation')
+    lottie.remove()
+  }
+  , time)
+}
+// This actually gets a name suggestion, but does not set it, should probably rename it
+const setRandomName = (startLetter) => {
+  let shortList = kittenLibrary.neutralNames
+  if (kitten.gender === 'girl') {
+    shortList = shortList.concat(kittenLibrary.girlNames)
+  } else if (kitten.gender === 'boy') {
+    shortList = shortList.concat(kittenLibrary.boyNames)
+  } 
+  let matchList = shortList.filter(name => {
+    let firstChar = name.charAt(0)
+    return firstChar.toLowerCase() === startLetter.toLowerCase() // Not really working 
+  })
+  if (!matchList.length) {
+    setRandomNameSuccess = false
+    matchList = shortList
+  } else {
+    setRandomNameSuccess = true
+  }
+  let name = matchList[Math.floor(Math.random() * matchList.length)]
+  return name
```

Comment:
You do not need to save a variable just to return it 😺 
`return matchList[Math.floor(Math.random() * matchList.length)]`